metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy

```diff
@@ -1083,6 +1083,66 @@ class MetacatSmokeSpec extends Specification {
         's3-mysql-db'                   | 'smoke_db6'  | 'part'    | ['test', 'unused'] as Set<String> | false
     }
 
+    @Unroll
+    def "Test tags for catalogName, databaseName and tableName"() {
+        when:
+        createTable(catalogName, databaseName, tableName)
+        def catalog = QualifiedName.ofCatalog(catalogName)
+        tagApi.setTags(TagCreateRequestDto.builder().name(catalog).tags(tags).build())
+        if (repeat) {
+            tagApi.setTags(TagCreateRequestDto.builder().name(catalog).tags(tags).build())
+        }
+
+        def database = QualifiedName.ofDatabase(catalogName, databaseName)
+        tagApi.setTags(TagCreateRequestDto.builder().name(database).tags(tags).build())
+
+        def table = QualifiedName.ofTable(catalogName, databaseName, tableName)
+        tagApi.setTags(TagCreateRequestDto.builder().name(table).tags(tags).build())
+
+        def catalogDto = api.getCatalog(catalogName)
+        def databaseDto = api.getDatabase(catalogName, databaseName, true, false)
+        def tableDto = api.getTable(catalogName, databaseName, tableName, true, true, false)
+
+        def ret = tagApi.search('test_tag', null, null, null)
+        def ret2 = tagApi.list(['test_tag'] as Set<String>, null, null, null, null)
+        def ret3 = tagApi.list(['never_tag'] as Set<String>, tags as Set<String>, null, null, null)
+
+        tagApi.removeTags(TagRemoveRequestDto.builder().name(catalog).tags([]).deleteAll(true).build())
+        tagApi.removeTags(TagRemoveRequestDto.builder().name(database).tags([]).deleteAll(true).build())
+        tagApi.removeTags(TagRemoveRequestDto.builder().name(table).tags([]).deleteAll(true).build())
+
+        def ret_new = tagApi.search('test_tag', null, null, null)
+        def ret2_new = tagApi.list(['test_tag'] as Set<String>, null, null, null, null)
+
+        then:
+        metacatJson.convertValue(catalogDto.getDefinitionMetadata().get('tags'), Set.class) == tags as Set<String>
+        metacatJson.convertValue(databaseDto.getDefinitionMetadata().get('tags'), Set.class) == tags as Set<String>
+        metacatJson.convertValue(tableDto.getDefinitionMetadata().get('tags'), Set.class) == tags as Set<String>
+
+        assert ret.size() == 3
+        assert ret2.size() == 3
+        assert ret3.size() == 0
+
+        assert ret_new.size() == 0
+        assert ret2_new.size() == 0
+
+        cleanup:
+        api.deleteTable(catalogName, databaseName, tableName)
```

Comment:
Does this cleanup the metadata/tags too?

Reply:
You want to include these databses in metacat.definition.metadata.delete.enableDeleteForQualifiedNames option in docker-compose.yml so that the metadata gets deleted too.

Reply:
why delete the metadata of them as part of the tag tests? For testing cleanup purpose?

Reply:
yes, for test cleanup.